### ISSUE TITLE
False positive: value is immediatly overwritten

### ISSUE THREAD

**Issue by Azenet** [kalessil/phpinspectionsea#1542]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended) 4.0.3 |
| Language level | PHP 7.3                        |

Current behaviour
--
Example code:
```php
<?php
$a   = [1, 2, 3];
$var = $result = null;

foreach ($a as $x) {
	if ($x === 2) {
		$result = $var;
		$var    = null;
	}

	$var = $x; // inspection here
}
```
Inspection is triggered on the last line, however the variable is used to mark the previous value for the next foreach loop and is used in the conditional.

Expected behaviour
--
The inspection should be placed on the `$var = null` assignment instead.

Note
--
Issue is maybe similar to #729.

Environment details
---
PhpStorm 2019.3.4
Build #PS-193.6911.26, built on March 18, 2020
Runtime version: 1.8.0_242-b08 amd64
VM: OpenJDK 64-Bit Server VM by Oracle Corporation
Linux 5.6.2-arch1-2-1khz

**Comment by kalessil** [comment]
Thank you for reporting @Azenet, I'll look into it.

**Comment by kalessil** [comment]
Checked: the inspection reports correctly (null assignment is conditional).
